### ISSUE TITLE
ValueError: unconverted data remains: .539842

### ISSUE THREAD

**Issue by mrichman** [achillesrasquinha/pipupgrade#115]
```
Checking...
2021-09-08 12:21:43,688 | INFO | Fetching installed packages for /usr/local/bin/pip...
2021-09-08 12:21:43,688 | INFO | Executing command: /usr/local/bin/pip list --outdated --format json
2021-09-08 12:21:43,691 | INFO | Fetching installed packages for /usr/local/opt/python@3.9/bin/pip3...
2021-09-08 12:21:43,691 | INFO | Executing command: /usr/local/opt/python@3.9/bin/pip3 list --outdated --format json
2021-09-08 12:21:55,196 | INFO | 5 packages found for /usr/local/opt/python@3.9/bin/pip3.
2021-09-08 12:21:55,532 | INFO | 5 packages found for /usr/local/bin/pip.
2021-09-08 12:21:57,392 | INFO | Initializing Package {'name': 'click', 'version': '7.1.2', 'latest_version': '8.0.1', 'latest_filetype': 'wheel'} of type <class 'dict'>...
2021-09-08 12:21:57,392 | INFO | Fetching package click information from DB...
2021-09-08 12:21:57,394 | INFO | Initializing Package {'name': 'dnspython', 'version': '1.16.0', 'latest_version': '2.1.0', 'latest_filetype': 'wheel'} of type <class 'dict'>...
2021-09-08 12:21:57,394 | INFO | Fetching package dnspython information from DB...
2021-09-08 12:21:57,401 | INFO | Initializing Package {'name': 'gevent', 'version': '20.12.1', 'latest_version': '21.8.0', 'latest_filetype': 'wheel'} of type <class 'dict'>...
2021-09-08 12:21:57,401 | INFO | Fetching package gevent information from DB...
2021-09-08 12:21:57,401 | INFO | Initializing Package {'name': 'setuptools', 'version': '57.4.0', 'latest_version': '58.0.3', 'latest_filetype': 'wheel'} of type <class 'dict'>...
2021-09-08 12:21:57,402 | INFO | Fetching package setuptools information from DB...
2021-09-08 12:21:57,402 | INFO | Initializing Package {'name': 'watchdog', 'version': '1.0.2', 'latest_version': '2.1.5', 'latest_filetype': 'wheel'} of type <class 'dict'>...
2021-09-08 12:21:57,403 | INFO | Fetching package watchdog information from DB...
2021-09-08 12:21:57,683 | INFO | Initializing Package {'name': 'click', 'version': '7.1.2', 'latest_version': '8.0.1', 'latest_filetype': 'wheel'} of type <class 'dict'>...
2021-09-08 12:21:57,683 | INFO | Fetching package click information from DB...
2021-09-08 12:21:57,698 | INFO | Initializing Package {'name': 'dnspython', 'version': '1.16.0', 'latest_version': '2.1.0', 'latest_filetype': 'wheel'} of type <class 'dict'>...
2021-09-08 12:21:57,699 | INFO | Fetching package dnspython information from DB...
2021-09-08 12:21:57,700 | INFO | Initializing Package {'name': 'gevent', 'version': '20.12.1', 'latest_version': '21.8.0', 'latest_filetype': 'wheel'} of type <class 'dict'>...
2021-09-08 12:21:57,700 | INFO | Fetching package gevent information from DB...
2021-09-08 12:21:57,701 | INFO | Initializing Package {'name': 'setuptools', 'version': '57.4.0', 'latest_version': '58.0.3', 'latest_filetype': 'wheel'} of type <class 'dict'>...
2021-09-08 12:21:57,701 | INFO | Fetching package setuptools information from DB...
2021-09-08 12:21:57,702 | INFO | Initializing Package {'name': 'watchdog', 'version': '1.0.2', 'latest_version': '2.1.5', 'latest_filetype': 'wheel'} of type <class 'dict'>...
2021-09-08 12:21:57,702 | INFO | Fetching package watchdog information from DB...

concurrent.futures.process._RemoteTraceback:
"""
concurrent.futures.process._RemoteTraceback:
"""
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/process.py", line 243, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/process.py", line 202, in _process_chunk
    return [fn(*args) for args in chunk]
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/process.py", line 202, in <listcomp>
    return [fn(*args) for args in chunk]
  File "/usr/local/lib/python3.9/site-packages/pipupgrade/model/package.py", line 129, in __init__
    time_difference	= to_datetime(res["_updated_at"]) + timedelta(seconds = cache_timeout)
  File "/usr/local/lib/python3.9/site-packages/pipupgrade/model/package.py", line 74, in to_datetime
    return datetime.strptime(string, "%Y-%m-%d %H:%M:%S")
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/_strptime.py", line 568, in _strptime_datetime
    tt, fraction, gmtoff_fraction = _strptime(data_string, format)
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/_strptime.py", line 352, in _strptime
    raise ValueError("unconverted data remains: %s" %
ValueError: unconverted data remains: .539842
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/process.py", line 243, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/process.py", line 202, in _process_chunk
    return [fn(*args) for args in chunk]
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/process.py", line 202, in <listcomp>
    return [fn(*args) for args in chunk]
  File "/usr/local/lib/python3.9/site-packages/pipupgrade/commands/helper.py", line 331, in get_registry_from_pip
    registry     = Registry(source = pip_path, packages = packages,
  File "/usr/local/lib/python3.9/site-packages/pipupgrade/model/registry.py", line 119, in __init__
    for package in pool.imap_unordered(partial(Package, **args), packages):
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/process.py", line 559, in _chain_from_iterable_of_lists
    for element in iterable:
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/_base.py", line 608, in result_iterator
    yield fs.pop().result()
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/_base.py", line 445, in result
    return self.__get_result()
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/_base.py", line 390, in __get_result
    raise self._exception
ValueError: unconverted data remains: .539842
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/pipupgrade/commands/__init__.py", line 77, in command
    return _command(**ARGUMENTS)
  File "/usr/local/lib/python3.9/site-packages/pipupgrade/commands/__init__.py", line 229, in _command
    for registry in pool.imap_unordered(
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/process.py", line 559, in _chain_from_iterable_of_lists
    for element in iterable:
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/_base.py", line 608, in result_iterator
    yield fs.pop().result()
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/_base.py", line 445, in result
    return self.__get_result()
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/_base.py", line 390, in __get_result
    raise self._exception
ValueError: unconverted data remains: .539842
```

**Comment by walksonair** [comment]
Same for me... was just coming to post but here it is..

**Comment by duduche59** [comment]
idem : 

> concurrent.futures.process._RemoteTraceback: 
"""
concurrent.futures.process._RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/usr/lib/python3.8/concurrent/futures/process.py", line 239, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
  File "/usr/lib/python3.8/concurrent/futures/process.py", line 198, in _process_chunk
    return [fn(*args) for args in chunk]
  File "/usr/lib/python3.8/concurrent/futures/process.py", line 198, in <listcomp>
    return [fn(*args) for args in chunk]
  File "/usr/local/lib/python3.8/dist-packages/pipupgrade/model/package.py", line 129, in __init__
    time_difference	= to_datetime(res["_updated_at"]) + timedelta(seconds = cache_timeout)
  File "/usr/local/lib/python3.8/dist-packages/pipupgrade/model/package.py", line 74, in to_datetime
    return datetime.strptime(string, "%Y-%m-%d %H:%M:%S")
  File "/usr/lib/python3.8/_strptime.py", line 568, in _strptime_datetime
    tt, fraction, gmtoff_fraction = _strptime(data_string, format)
  File "/usr/lib/python3.8/_strptime.py", line 352, in _strptime
    raise ValueError("unconverted data remains: %s" %
ValueError: unconverted data remains: .108715
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3.8/concurrent/futures/process.py", line 239, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
  File "/usr/lib/python3.8/concurrent/futures/process.py", line 198, in _process_chunk
    return [fn(*args) for args in chunk]
  File "/usr/lib/python3.8/concurrent/futures/process.py", line 198, in <listcomp>
    return [fn(*args) for args in chunk]
  File "/usr/local/lib/python3.8/dist-packages/pipupgrade/commands/helper.py", line 331, in get_registry_from_pip
    registry     = Registry(source = pip_path, packages = packages,
  File "/usr/local/lib/python3.8/dist-packages/pipupgrade/model/registry.py", line 119, in __init__
    for package in pool.imap_unordered(partial(Package, **args), packages):
  File "/usr/lib/python3.8/concurrent/futures/process.py", line 484, in _chain_from_iterable_of_lists
    for element in iterable:
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 619, in result_iterator
    yield fs.pop().result()
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 444, in result
    return self.__get_result()
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 389, in __get_result
    raise self._exception
ValueError: unconverted data remains: .108715
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/pipupgrade/commands/__init__.py", line 77, in command
    return _command(**ARGUMENTS)
  File "/usr/local/lib/python3.8/dist-packages/pipupgrade/commands/__init__.py", line 229, in _command
    for registry in pool.imap_unordered(
  File "/usr/lib/python3.8/concurrent/futures/process.py", line 484, in _chain_from_iterable_of_lists
    for element in iterable:
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 619, in result_iterator
    yield fs.pop().result()
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 444, in result
    return self.__get_result()
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 389, in __get_result
    raise self._exception
ValueError: unconverted data remains: .108715

An error occured while performing the above command. This could be an issue with
"pipupgrade". Kindly post an issue at https://github.com/achillesrasquinha/pipupgrade/issue

**Comment by machinexa2** [comment]
https://github.com/achillesrasquinha/pipupgrade/issues/116

**Comment by meesha7** [comment]
This issue happens in v 1.10.0, while 1.9.0 is working

**Comment by meesha7** [comment]
If you delete `~/.config/pipupgrade` it starts working, so I'm guessing it's [this line](https://github.com/achillesrasquinha/pipupgrade/blob/master/src/pipupgrade/model/package.py#L129), maybe the DB entry's `_updated_at` changed format?

**Comment by FernandoMiguel** [comment]
> If you delete `~/.config/pipupgrade` it starts working, so I'm guessing it's [this line](https://github.com/achillesrasquinha/pipupgrade/blob/master/src/pipupgrade/model/package.py#L129), maybe the DB entry's `_updated_at` changed format?

thanks @meesha7

**Comment by achillesrasquinha** [comment]
Fixed with 1.10.1
